jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -48,4 +50,11 @@
     private static IWorkspaceRoot getWorkspaceRoot() {
         return ResourcesPlugin.getWorkspace().getRoot();
     }
+
+    // TODO Use ProjectUtils.getWorkspaceInvisibleProjectName directly when the language server is released.
+    private static String getWorkspaceInvisibleProjectName(IPath workspacePath) {
+        String fileName = workspacePath.toFile().getName();
+        String projectName = fileName + "_" + Integer.toHexString(workspacePath.toPortableString().hashCode());
```

Comment:
return fileName + "_" + Integer.toHexString(workspacePath.toPortableString().hashCode())

Reply:
string.format

Reply:
i think current style is more readable than string.format.